src/main/java/kr/syeyoung/dungeonsguide/Main.java

```diff
@@ -33,110 +39,229 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import java.io.*;
-import java.lang.reflect.InvocationTargetException;
 import java.net.URL;
-import java.security.*;
-import java.security.cert.CertificateException;
-import java.security.spec.InvalidKeySpecException;
+import java.util.List;
 
 @Mod(modid = Main.MOD_ID, version = Main.VERSION)
-public class Main
-{
+public class Main {
     public static final String MOD_ID = "skyblock_dungeons_guide";
-    public static final String VERSION = "1.0";
+    public static final String VERSION = "3.8";
+    @Getter
+    private static boolean firstTimeUsingDG = false;
+    @Getter
+    static File configDir;
 
-    private static Main main;
-
-    private DGInterface dgInterface;
 
     private boolean isLoaded = false;
     private Throwable cause;
     private String stacktrace;
     private boolean showedError = false;
 
+    Logger logger = LogManager.getLogger("DG-main");
+
+    YoMamaOutdated yoMamaOutdated;
 
+    static public File getConfigDir() {
+        return configDir;
+    }
 
     @EventHandler
-    public void initEvent(FMLInitializationEvent initializationEvent)
-    {
+    public void initEvent(final FMLInitializationEvent initializationEvent) {
+//        if(yoMamaOutdated == null || yoMamaOutdated.isUsingOutdatedDg) return;
         MinecraftForge.EVENT_BUS.register(this);
-        if (dgInterface != null) {
-            main = this;
-            try {
-                dgInterface.init(initializationEvent);
-            } catch (Exception e) {
-                cause = e;
-                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-                PrintStream printStream = new PrintStream(byteArrayOutputStream);
-                e.printStackTrace(printStream);
-                stacktrace = new String(byteArrayOutputStream.toByteArray());
+        try {
+            logger.info("init-ing DungeonsGuide");
+            DungeonsGuide.getDungeonsGuide().init();
+        } catch (Exception e) {
+            cause = e;
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            PrintStream printStream = new PrintStream(byteArrayOutputStream);
+            e.printStackTrace(printStream);
+            stacktrace = byteArrayOutputStream.toString();
 
-                e.printStackTrace();
-            }
+            e.printStackTrace();
         }
     }
 
+    private boolean showedStartUpGuide;
+
     @SubscribeEvent(priority = EventPriority.HIGHEST)
     public void onGuiOpen(GuiOpenEvent guiOpenEvent) {
-        if (!showedError && !isLoaded && guiOpenEvent.gui instanceof GuiMainMenu) {
+        if(!(guiOpenEvent.gui instanceof GuiMainMenu)) return;
+        if (!showedError && !isLoaded) {
             guiOpenEvent.gui = new GuiLoadingError(cause, stacktrace, guiOpenEvent.gui);
             showedError = true;
+            return;
+        }
+        if(!showedStartUpGuide){
```

Comment:
Should be moved to DugneonsGuide.java

Main class exists for the purpose of loading the mod